scripts/s3migrate.sh

```diff
@@ -0,0 +1,388 @@
+                echo ""
+                read -r -p "Would you like to use these credentials? [y/N] " response
+                    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                        echo "Setting detected credentials"
+                        export AWS_ACCESS_KEY_ID=${access_key_id//\"}
+                        export AWS_SECRET_ACCESS_KEY=${secret_access_key//\"}
+                        return
+                    else
+                        credCheck
+                    fi
+            else
+                echo "Minio will use whatever credentials you've configured it with."
+                echo "If those credentials don't match your aws credentials file, you"
+                echo "must specify those custom credentials."
+                credCheck
+            fi
+        done
+    else
+        credCheck
+    fi
+}
+
+cleanYoself() {
+    echo ""
+    echo "==========================================================="
+    read -r -p "Would you like to clean all local artifact storage locations? [y/N]" response
+        if [[ "${response}" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            echo "==========================================================="
+            echo "###########################################################"
+            echo "### CAUTION: This is a destructive operation and cannot ###"
+            echo "### be undone or reversed. It is 100% entirely permanent ##"
+            echo "###########################################################"
+            echo "==========================================================="
+            read -r -p "Are you certain you would like to clean local storage? [y/N]" response2
+                if [[ "${response2}" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Removing all content at /hab/svc/builder-api/data/pkgs/"
+                    rm -rf "/hab/svc/builder-api/data/pkgs/"
+                    echo "Cleanup complete!"
+                else
+                    echo "Exiting without removing locally stored artifacts"
+                    exit 0
+                fi
+        else
+            echo "Exiting without removing locally stored artifacts"
+        fi
+}
+
+artifact_dir="/hab/svc/builder-api/data/pkgs"
+artifacts=()
+
+if [[ -z ${1:-} ]]; then
+    echo "Invalid Argument. Argument must be either 'minio' or 'aws'"
+    exit 1
+fi
+
+case ${1} in
+    'minio')
+        echo "Starting migration to minio instance."
+        export s3type="minio"
+        installDeps
+        welcome
+        setRegion
+        setBucket
+        getHarts "${artifact_dir}"
+        time uploadHarts
+    ;;
+    'aws')
+        echo "Starting migration to AWS S3."
+        export s3type="aws"
+        installDeps
+        welcome
+        setRegion
+        setBucket
+        getHarts "${artifact_dir}"
+        time uploadHarts
+    ;;
+    *) echo "Invalid argument. Arg must be 'minio' or 'aws'"
+        exit 1
+    ;;
+esac
+
+cleanYoself
```

Comment:
Given that a significant amount of time was just spent linting all of our existing shell scripts through `shellcheck`, and correcting the errors it found, I think it'd be worthwhile to do that to this shell script as well.  I haven't run `shellcheck` on it, but I can see just from a visual inspection that it's not in line with the way our other scripts will be after https://github.com/habitat-sh/habitat/pull/4974 merges.

Reply:
Hmm, @eeyun ran shellcheck and didn't get any errors..

Reply:
My mistake then